### PR TITLE
Enable setting alluxio-fuse options for docker image

### LINKED ISSUE #9441: It's about double time cost using fuse comparing copying from disk
**Alluxio Version:**
2.0.0

**Describe the bug**

The time of copying data in fuse(cached) is about 15 minutes, but copying data directly from disk is 7 minutes. It's about double cost comparing copying from disk. Any way to optimize?

The data is already cached in the alluxio worker in the same node,  and copy the 

1. Deploy master, workers and fuse in kubernetes

[k8s.zip](https://github.com/Alluxio/alluxio/files/3343410/k8s.zip)

2. And go to the node(iZhp3bku0ru8vuxq08lorwZ),  copy whole the directory from alluxio fuse to local directory

```
time cp -r /alluxio-fuse/training-data/images /test

real    133m33.728s
user    0m1.518s
sys     2m58.588s
```

3. After that, check the persistent status. All the data are in persistent.

```
bash-4.4# /opt/alluxio/bin/alluxio fs ls /training-data/images| wc -l
1152
bash-4.4# /opt/alluxio/bin/alluxio fs ls /training-data/images| grep  PERSIST|wc -l
1152
bash-4.4# /opt/alluxio/bin/alluxio fs ls /training-data/images| grep -v PERSIST|wc -l
0
```

3. I've checked the data is fully loaded into alluxio, in node iZhp3bku0ru8vuxq08lorwZ

```
bash-4.4# /opt/alluxio/bin/alluxio fsadmin report ufs
Alluxio under storage system information:
oss://imagenet-huabei5/images      on  /training-data/images  (oss, capacity=-1B, used=-1B, read-only, not shared, properties={fs.oss.accessKeySecret=******, fs.oss.accessKeyId=******, fs.oss.endpoint=oss-cn-internal.aliyuncs.com})
/opt/alluxio-2.0.0/underFSStorage  on  /                      (local, capacity=4843.27GB, used=-1B(0%), not read-only, not shared, properties={})
```

```
bash-4.4# /opt/alluxio/bin/alluxio fsadmin report capacity
Capacity information for all workers:
    Total Capacity: 2400.00GB
        Tier: MEM  Size: 400.00GB
        Tier: SSD  Size: 2000.00GB
    Used Capacity: 167.41GB
        Tier: MEM  Size: 109.58GB
        Tier: SSD  Size: 57.82GB
    Used Percentage: 6%
    Free Percentage: 94%

Worker Name                  Last Heartbeat   Storage       Total            MEM           SSD
iZhp3bku0ru8vuxq08loruZ      0                capacity      600.00GB         100.00GB      500.00GB
                                              used          0B (0%)          0B            0B
iZhp3bku0ru8vuxq08lorvZ      0                capacity      600.00GB         100.00GB      500.00GB
                                              used          0B (0%)          0B            0B
iZhp3bku0ru8vuxq08lorxZ      0                capacity      600.00GB         100.00GB      500.00GB
                                              used          23.74GB (3%)     23.74GB       0B
iZhp3bku0ru8vuxq08lorwZ      0                capacity      600.00GB         100.00GB      500.00GB
                                              used          143.67GB (23%)   85.84GB       57.82GB
```

4. I went to the same node to copy data, it took about 15minutes

```
time cp -r /alluxio-fuse/training-data/images /test

real    15m56.379s
user    0m1.178s
sys     2m45.012s
```

5. Check the metrics,  I can make sure it goes through socket

```
bash-4.4# /opt/alluxio/bin/alluxio fsadmin report metrics
Total IO:
    Short-circuit Read                                         0B
    Short-circuit Read (Domain Socket)                   593.25GB
    From Remote Instances                                      0B
    Under Filesystem Read                                      0B
    Alluxio Write                                              0B
    Alluxio Write (Domain Socket)                              0B
    Under Filesystem Write                                     0B

Total IO Throughput (Last Minute):
    Short-circuit Read                                         0B
    Short-circuit Read (Domain Socket)                   111.21MB
    From Remote Instances                                      0B
    Under Filesystem Read                                      0B
    Alluxio Write                                              0B
    Alluxio Write (Domain Socket)                              0B
    Under Filesystem Write                                     0B
```

6. But I copied the data on the disks, it's about 7 minutes

```
time cp -r /test/* /var/lib/docker/test/

real    7m24.108s
user    0m0.734s
sys     2m20.118s
```

Are there any wrong with my configuration?

**To Reproduce**
Steps to reproduce the behavior (as minimally and precisely as possible)

**Expected behavior**

**Urgency**
Describe the impact and urgency of the bug.

**Additional context**

### PR DESCRIPTION
**Pull request by zrss**
fixes #9441

add a way for setting extra options for alluxio-fuse through args

### CODE REVIEW

**Comment by apc999** [comment]
can you provide more details on the issue you are fixing? It is not clear to me yet.

**Comment by zrss** [comment]
oh, i'd like to set some extra option for fuse, such as

```
-o allow_other,kernel_cache,auto_cache,entry_timeout=120,attr_timeout=120
```

currently, i have to modify the entripoint.sh shell rather than the args in alluxio-fuse.yaml

**Comment by madanadit** [comment]
Thanks @zrss. Can you add a description to at the top of this thread with how these options improve performance?

File: integration/docker/entrypoint.sh [hunk 1]
```diff
@@ -129,7 +129,7 @@ case ${service,,} in
   fuse)
     # Unmount first if cleanup failed and ignore error
     ! integration/fuse/bin/alluxio-fuse unmount /alluxio-fuse
-    integration/fuse/bin/alluxio-fuse mount -o allow_other /alluxio-fuse /
+    integration/fuse/bin/alluxio-fuse mount ${options} /alluxio-fuse /
```
Comment (LuQQiu) [inline integration/docker/entrypoint.sh]: Agreed that we can allow passing mount point options through docker entrypoint, but using existing `${options}` seems a little confusing. For all other commands, options refer to `--no-format`, but for fuse, it refers to fuse mount options. Maybe we  can pass in `fuse --fuse-opts=<fuse options separated by comma>`? Could you also update the usage docs?

Comment (zrss) [reply integration/docker/entrypoint.sh]: will do

**Comment by zrss** [comment]
@madanadit unfortunately, it can't work well without `direct_io` option in alluxio (read a tar.gz 200MB will be hang, but a small json file works well) ... the other fuse option i read at here [mount.fuse](http://man7.org/linux/man-pages/man8/mount.fuse.8.html)

File: integration/docker/entrypoint.sh (hunk 1 above)
Comment (zrss) [reply integration/docker/entrypoint.sh]: done

and as i change the entrypoint.sh so the image of alluxio-fuse have to release too ...

**Comment by gpang** [comment]
@zrss Please target this improvement to `master` branch first, and then cherry-pick to other release branches. Thanks!

**Comment by zrss** [comment]
@gpang @LuQQiu done

File: integration/docker/entrypoint.sh [hunk 2]
```diff
@@ -13,19 +13,20 @@
...
-  echo -e " proxy                        \t Start Alluxio proxy"
-  echo -e " fuse                         \t Start Alluxio fuse"
+  echo -e " master [--no-format]                                  \t Start Alluxio master. If --no-format is specified, do not format"
+  echo -e " master-only [--no-format]                             \t Start Alluxio master w/o job master. If --no-format is specified, do not format"
+  echo -e " worker [--no-format]                                  \t Start Alluxio worker. If --no-format is specified, do not format"
+  echo -e " worker-only [--no-format]                             \t Start Alluxio worker w/o job worker. If --no-format is specified, do not format"
+  echo -e " job-master                                            \t Start Alluxio job master"
+  echo -e " job-worker                                            \t Start Alluxio job worker"
+  echo -e " proxy                                                 \t Start Alluxio proxy"
+  echo -e " fuse [--fuse-opts=<fuse options separated by comma>]  \t Start Alluxio fuse with options for FUSE file systems"
```
Comment (apc999) [inline integration/docker/entrypoint.sh]: `fuse [--fuse-opts=opt1,...]  \t Start Alluxio FUSE file system, option --fuse-opts expects a list of fuse options separated by comma"`

**Comment by apc999** [comment]
One minor comment left. Otherwise LGTM

File: integration/docker/entrypoint.sh (hunk 2 above)
Comment (zrss) [reply integration/docker/entrypoint.sh]: done

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**